overlord/snapstate/aliasesv2.go

```diff
@@ -568,7 +568,7 @@ func manualAlias(info *snap.Info, curAliases map[string]*AliasTarget, target, al
 		} else {
 			reason = fmt.Sprintf("target application %q is a daemon", target)
 		}
-		return nil, fmt.Errorf("cannot enable alias %q for %q, %s", alias, info.Name(), reason)
+		return nil, fmt.Errorf("cannot enable alias %q for %q, %s", alias, info.InstanceName(), reason)
```

Comment:
CC @pedronis are aliases in any way reliant upon parsing the snap name? Would the introduction of `_` affect them?

Reply:
they parse app names but that's not the problem, but the alias code has probably assumptions about name<->sna-id being unique on a system so far

Reply:
Added more TODOs in places where `snapsup.RealName` is used. The rest will need careful walkthough.